framework/py/flwr/common/record/array.py

```diff
@@ -248,3 +256,32 @@ def numpy(self) -> NDArray:
         # Source: https://numpy.org/doc/stable/reference/generated/numpy.load.html
         ndarray_deserialized = np.load(bytes_io, allow_pickle=False)
         return cast(NDArray, ndarray_deserialized)
+
+    def serialize(
+        self, refs_dict: Optional[dict[str, str]] = None
+    ) -> Union[bytes, str]:  # noqa: D102
+        if refs_dict:
+            raise NotImplementedError()
+        array_proto = ArrayProto(
+            dtype=self.dtype,
+            shape=self.shape,
+            stype=self.stype,
+            data=self.data,
+        )
+
+        obj_content = array_proto.SerializeToString(deterministic=True)
+        full_serialized = add_header_to_object_content(
+            object_content=obj_content, cls=self
+        )
+        return full_serialized, get_object_id(full_serialized)
```

Comment:
Proposal to align the naming:
```suggestion
        obj_body = array_proto.SerializeToString(deterministic=True)
        obj_content = add_header_to_object_content(
            object_content=obj_body, cls=self
        )
        return obj_content, get_object_id(obj_content)
```